pkg/proc/target.go

```diff
@@ -26,8 +28,25 @@ type Target struct {
 	gcache goroutineCache
 }
 
+// StopReason describes the reason why the target process is stopped.
+// A process could be stopped for multiple simultaneous reason, in which
+// case only one will be reported.
+type StopReason uint8
+
+const (
+	StopUnknown             StopReason = iota
+	StopLaunched                       // The process was just launched
+	StopAttached                       // The debugger stopped the process after attaching
+	StopExited                         // The target proces terminated
+	StopBreakpoint                     // The target process hit one or more software breakpoints
```

Comment:
Is it not worth differentiating what kind of breakpoint this was (function, data condition, etc)?

Reply:
We don't have data breakpoints, when we do we'll do that. Differentiating between function breakpoints and breakpoints on things other than the first line of a function would be complicated and IMHO not very useful.